src/ds/d500/d500-options.cpp

```diff
@@ -50,4 +50,33 @@ namespace librealsense
 
         return temperature;
     }
+    temperature_xu_option::temperature_xu_option(const std::weak_ptr<uvc_sensor>& ep, 
+        platform::extension_unit xu, uint8_t id, 
+        std::string description, bool allow_set_while_streaming)
+        : uvc_xu_option<int16_t>(ep, xu, id, description, allow_set_while_streaming)
+    {
+        // defining the parsing modifier, to be used on the calls for query method
+        _parsing_modifier = [](const int16_t read_value) {
+            return static_cast<float>(read_value) / 10.f;
+        };
+    }
+
+    float temperature_xu_option::query() const
+    {
+        float temperature = -1;
+        try 
+        {
+            temperature = uvc_xu_option<int16_t>::query();
+        }
+        catch (...)
```

Comment:
Why not leaving the original exception?

![image](https://github.com/IntelRealSense/librealsense/assets/64067618/a2f1f203-7da7-486c-b4ff-0816c69f593f)

Code will be cleaner.

Reply:
As discussed, remove the try/catch

Reply:
ok